File: Daily Coding Practice/30 -day coding challenge -- May/4. Compliment number.py
classs Solution:
    def compliment(num):
        bit_count = 0
        n = num
        while n > 0:
            bit_count += 1
            n = n >> 1

        all_bits_set = pow(2, bit_count) - 1
        # all_bits_set = (1 << bit_count) - 1
        # use bit manipulation to solve it

        return num ^ all_bits_set

# time complexity o(b)
# space complexity o(1)

File: Daily Coding Practice/30 -day coding challenge -- May/4. Compliment number.cpp
using namespace std;

#include<iostream>
#include<math.h>

class CalculateComplement{
public:
    static int bitwiseComplement(int num){
    int bitCount = 0;
    int n = num;
    while (n > 0){
        bitCount ++;
        n = n >> 1;
    }
    int all_bits_set = pow(2, bitCount) -1;
    return num ^ all_bits_set;
    }
};